plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-orders.php

```diff
@@ -662,5 +662,23 @@ public function search_custom_fields( $wp ) {
 			// Search by found posts.
 			$wp->query_vars['post__in'] = array_merge( $post_ids, array( 0 ) );
 		}
+
+		if ( isset( $_GET['m'] ) ) {
+			$date_option = get_option( 'woocommerce_date_type', 'date_paid' );
+			$date_query  = wc_clean( wp_unslash( $_GET['m'] ) );
+
+			unset( $wp->query_vars['m'] );
+
+			if ( 'date_paid' === $date_option || 'date_completed' === $date_option ) {
```

Comment:
It took me a little while to figure out why you had to specifically add the date filters for these, but then realized that we should check for the `date_paid` or `date_completed` dates instead.
It might be worth adding a clarifying comment.
Although I actually think we should add support for an additional `query_vars` here, we can call it `order_date_type` or something like that. As this also breaks the normal view of the order list, and given this isn't part of Analytics, I don't think we should be reflecting the data within this table dependant on what a setting in Analytics is set to.
Instead we are better off supporting the additional query type, so we can add that to the links from Analytics.